### ISSUE TITLE
Grouping Characters

### ISSUE THREAD

**Issue by Biep** [olivierkes/manuskript#1141]
Currently, there are three hard-coded groups for characters: *Main*, *Secondary*, and *Minor*.
For me, those labels are not very useful - especially the *Minor* one.  I get a huge unordered list of people there, and finding the proper person ("Who again was the math teacher of character X - I know she has been mentioned somewhere, and I need to be consistent") is a lot of work.

Being able to build a tree - a bit like the one for "World" - would be enormously helpful.  That way I can group people, fold classes that currently are irrelevant, and quickly zoom in on the person I know.

Both sections could use the same code, so that any enhancement or bug fix at the one place will immediately happen at the other too.  That way the total code base for the program would actually *decrease*.

**Comment by Biep** [comment]
By the way - I would have no problem with *Character* being just one of a set of kinds of items that can be added to the world.  If my main characters pass through a village, and meet the butcher there, it makes sense for me to describe that butcher as "inside" that village, so that if ever later the story touches that village again, I'll find the relevant characters to interact with ready in its description.

**Comment by TheJackiMonster** [comment]
I think one reason to separate code of world view and characters view would be that world items allows having child items. So they can be a topic as well as a specific item. I personally use them as locations which can include more precise other locations. For example a house inside a city inside a country. But at the same time these items can also describe objects.

With characters they are a lot more defined in Manuskript. I agree that those categories are oddly specific and not appropriate for any use case. I assume we could make those categories completely custom if we replace the input element to select the category with a drop down element. This is also something suggested in other issues.

**Comment by Biep** [comment]
I agree that for most stories, a character would not have child characters in the sense an item in the world can have child items.  But the *categories* can have child categories: the school population could be have subpopulations "staff" and "pupils", with the latter broken up in classes again, for instance.  And those categories would be a natural fit in the world.

**Comment by TheJackiMonster** [comment]
So you think we should have subgroups of groups as well? Because "staff" or "pupils" wouldn't be a character of course or do you think we should make one view arranging world items and characters at once? So you group characters under world items as sub-categories... because I assume that would work as well but maybe be more confusing to some users.

At the moment you can use Manuskript completely avoiding the world view in case characters are the only important part to your story for example.

**Comment by TheShadowOfHassen** [comment]
So how about instead of major secondary and minor we add the ability to make custom sub categories like "Police Officers" or "Good guys" Also have the categories be hid/ shown so if you have 100 characters you can only have open the reoccurring characters and not the  people you meet on the way.

**Comment by Biep** [comment]
Yes, that is what I should like to have.
And the categories would have their own information.  The Vulcans have pointy ears, for instance, and the Fulani would be nomads with specific habits.

**Comment by Biep** [comment]
Maybe leaf nodes should have a few toggles:
- whether they have intention (and thus can be actors)
- whether they are the object of contention (and thus trigger actors)
- whether they have a timeline (and thus may have different aspects at different points in the story).
- ..?

My village may have no timeline, but my car may - it moves, becomes increasingly erratic and unreliable, and eventually breaks down.  Being a valuable antique, it is also the object of contention, and drives the plot.
The daughter of the butcher is a flat character without intention (in the story), but is an object of contention between het many suitors.

**Comment by vextronx** [comment]
@Biep 
I think all it would take to implement this would be giving the character view the organization of the World view. The only other thing we'd need is making it possible to edit these categories' names and give them a description field. Or do you mean something more?

Maybe we could also make some cross-referencing with world objects and characters.

**Comment by TheShadowOfHassen** [comment]
I think we need to be careful about this because in my mind the whole idea of the characters tab is when I write a story with ten characters I can keep track of them without having to do to much work. Maybe what you can do is add/remove catagory heads and then you can link those to the world. I only need main/secondary/minor most of the time. I'd love to change it when I need more but most of the time I just need those three and the UI to stay out of my way as I go about writing.

**Comment by Biep** [comment]
I'd like to think of it the way a file folder explorer works: it can unfold and fold subcategories, and also zoom in to a specific category.  If one zooms in to "people", nothing else would be shown, just as when I explore a certain folders only it and its contents are shown.  That would cover your use case, wouldn't it?

**Comment by Biep** [comment]
> @Biep I think all it would take to implement this would be giving the character view the organization of the World view. The only other thing we'd need is making it possible to edit these categories' names and give them a description field. Or do you mean something more?
> 
> Maybe we could also make some cross-referencing with world objects and characters.

For those who want all their characters grouped together it would - but that is a (not uncommon) special case.  I like to have some characters grouped with the world they live in - the mentioned butcher would stay completely out of sight if I am not exploring the village she lives in, and when I **do** explore that village, its inhabitants will present itself to me and remind me of their existence - together with the picturesque mansion and the church with its vibrant teaching.

But having a simple hierarchy for characters would already go a big way in making the program (even) more useful.

**Comment by Biep** [comment]
I just saw [this](https://alternativeto.net/software/fantasia-archive/about/).  While I haven't looked at the program, the description seems to provide a lot of what I'd like to be able to do.  The predefined base categories could still be the ones Manuskript currently has, but the freedom to build custom categories would ensure that that would not be restrictive.

**Comment by Biep** [comment]
@kiwipen wrote:
> In your example I could replace 'Police Men' with various factions, and list major players in each one.

Yes - or in the `Acts` fragment @olivierkes uses there could be the Apostles, the Roman dignitaries, the converts, the Jewish establishment, and so on - with e.g. Paul changing from "Jewish establishment" to "Apostle" at some point in time.  Being able to model that is extremely useful in a complex story.

**Comment by Biep** [comment]
Not sure where to place this, so I'll repeat it at all the issues where it seems relevant.

I tried reordering characters the hard way - renumbering the files in the characters folder.
That led to the first time all characters appearing twice, first in the old sort order and then in the new - and I could delete one without deleting the other. The second time I started Manuskript all characters appeared only once again, and in the old order.
Where is that order kept? I cannot find a file, but if Manuskript knows it it must be somewhere - it clearly is not just the file numbering, because I changed that, yet the old order returned.
In characters, the files are back to their old numbering.

With a few hundred characters, being able to organise them is vital. I don't mind getting my hands dirty and editing files, but how should I go about it?
